prober/prober.go

```diff
@@ -574,7 +576,66 @@ func (p *Prober) RunHandler(w http.ResponseWriter, r *http.Request) error {
 		return tsweb.Error(respStatus, fmt.Sprintf("Probe failed: %s\n%s", err.Error(), stats), err)
 	}
 	w.WriteHeader(respStatus)
-	w.Write([]byte(fmt.Sprintf("Probe succeeded in %v\n%s", info.Latency, stats)))
+	fmt.Fprintf(w, "Probe succeeded in %v\n%s", info.Latency, stats)
+	return nil
+}
+
+type RunHandlerAllResponse struct {
+	Results map[string]RunHandlerResponse
+}
+
+func (p *Prober) RunAllHandler(w http.ResponseWriter, r *http.Request) error {
+	probes := make(map[string]*Probe)
+	p.mu.Lock()
+	for _, probe := range p.probes {
+		if !probe.IsContinuous() && probe.name != "derpmap-probe" {
+			probes[probe.name] = probe
+		}
+	}
+	p.mu.Unlock()
+
+	g, _ := errgroup.WithContext(r.Context())
```

Comment:
Question (non-blocking): Are you swallowing the context on purpose, to let all of the probes run to completion? Usually, errgroup is used to abort their goroutines on the first failure, but if you’re intending to run them all, can you please leave a comment to that effect?

```suggestion
	// Do not abort running probes just because one of them has failed.
	g, _ := errgroup.WithContext(r.Context())
```

Reply:
Yes I think the context is irrelevant here. I should have noted that on your first change request.

I think just the following is sufficient since we want all the probes to run.
```suggestion
	g := errgroup.Group{}
```